client/src/components/advancedSearch/TaskFilter.tsx

```diff
@@ -1,41 +1,242 @@
+    }
+  }
+
+  const buildFlattenedList = (tasks, level = 0) => {
+    return tasks.flatMap(task => {
+      const isTaskSelected = selectedItems?.some(
+        item => item.uuid === task.uuid
+      )
+      const isDescendantTaskSelected = selectedItems?.some(item =>
+        task.descendantTasks?.some(child => child.uuid === item.uuid)
+      )
+      const isAscendantTaskSelected = selectedItems
+        ?.filter(item => item.uuid !== task.uuid)
+        ?.some(item =>
+          task.ascendantTasks?.some(child => child.uuid === item.uuid)
+        )
+      const isCollapsed = !expandedItems.has(task.uuid)
+      const isSelected =
+        isTaskSelected || isAscendantTaskSelected
+          ? true
+          : isDescendantTaskSelected && isCollapsed
+            ? null
+            : false
+      const disabled = isAscendantTaskSelected
+      const taskWithLevel = { ...task, level, isSelected, disabled }
+      const children = expandedItems.has(task.uuid)
+        ? childrenMap.get(task.uuid) || []
+        : []
+      return [taskWithLevel, ...buildFlattenedList(children, level + 1)]
+    })
+  }
+
+  const topLevelItems = items.filter(task => !task.parentTask)
+  const flattenedItems = buildFlattenedList(topLevelItems)
+
+  const enhancedRenderRow = task => {
+    const hasChildren = task.childrenTasks?.length > 0
+    const isExpanded = expandedItems.has(task.uuid)
+    const isSelected = selectedItems?.some(item => item.uuid === task.uuid)
+
+    const handleToggleSelection = e => {
+      e.stopPropagation()
+      if (isSelected) {
+        handleRemoveItem(task)
+      } else {
+        handleAddItem(task)
+      }
+    }
+
+    const displayLabel = task.longName
+      ? `${task.shortName}: ${task.longName}`
+      : task.shortName
+    const padding = Math.min(task.level, 3) * 20 + (hasChildren ? 0 : 26)
+    const indentedLabel = (
+      <div
+        style={{
+          display: "flex",
+          flexDirection: "row",
+          alignItems: "center",
+          paddingLeft: padding,
+          gap: 10,
+          cursor: "auto"
+        }}
+      >
+        {hasChildren && (
+          <span
+            onClick={e => {
+              e.stopPropagation()
+              handleExpand(task)
+            }}
+            style={{ cursor: "pointer", pointerEvents: "all" }}
+          >
+            <Icon
+              icon={
+                isExpanded ? IconNames.CHEVRON_DOWN : IconNames.CHEVRON_RIGHT
+              }
+              size={IconSize.STANDARD}
+              color="#5f6b7c"
+            />
+          </span>
+        )}
+        {hasChildren ? (
+          <Icon
+            icon={isExpanded ? IconNames.FOLDER_OPEN : IconNames.FOLDER_CLOSE}
+            size={IconSize.STANDARD}
+            color="#5f6b7c"
+          />
+        ) : (
+          <Icon
+            icon={IconNames.DOCUMENT}
+            size={IconSize.STANDARD}
+            color="#5f6b7c"
```

Comment:
`color` prop can be removed (see https://github.com/NCI-Agency/anet/pull/5136#discussion_r2020927930 ).